src/services/movieService.ts

```diff
@@ -239,11 +243,40 @@ const getMovieCastById = async (movieId: number) => {
   };
 };
 
+const getMoviesByTitle = async (titlePart: string) => {
+  const movies = await prisma.movies.findMany({
+    where: {
+      title: {
+        contains: titlePart,
+        mode: "insensitive",
+      },
+    },
+    select: {
+      id: true,
+      title: true,
+      poster_path: true,
+      release_date: true,
+    },
+    take: 4,
+    orderBy: {
+      vote_average: "desc",
+    },
+  });
+
+  return movies.map(movie => ({
+    id: movie.id,
+    title: movie.title,
+    posterPath: movie.poster_path,
+    releaseDate: movie.release_date,
```

Comment:
En tiedä onko kovin olennaista alkaa näitä muuttujien nimiä vaihtamaan, mutta ei niin väliä mulle